test-dts/mount.d-test.ts

```diff
@@ -71,28 +82,43 @@ const AppWithArrayProps = {
   template: ''
 }
 
-// accept props
-wrapper = mount(AppWithArrayProps, {
-  props: { a: 'Hello' }
-})
-// vm is properly typed
-expectType<string>(wrapper.vm.a)
+// accept props - vm is properly typed
+expectType<string>(
+  mount(AppWithArrayProps, {
+    props: { a: 'Hello' }
+  }).vm.a
+)
 
 // can receive extra props
 // as they are declared as `string[]`
 mount(AppWithArrayProps, {
   props: { a: 'Hello', b: 2 }
 })
 
+// cannot receive extra props
+// if they pass use object inside
+expectError(
+  mount(
+    {
+      props: ['a']
+    },
```

Comment:
I'm happy to see that it throws an error, but I don't get how this is different from the test case with  `AppWithArrayProps` which still succeeds? Maybe add more explanation in the comment.
Ideally, shouldn't `mount(AppWithArrayProps, {  props: { a: 'Hello', b: 2 } })` also fail?

Reply:
Basically when you use `mount({ props: ['a']}) is basically the same as doing `defineComponent({props:['a']})`, the ThisType will be inferred as the `$vm`, resulting in:
```ts

mount({
  props: ['a']
})
// type
type MountTyped = {
  props: ['a'] 
} 


const Comp = {
  props: ['a']
}
//type
type CompTyped = {
  props: string[]
}
```

Basically using `mount/defineComponent` we maintain the actual tuple value, when using a variable with the options, typescript will resolve the tuple as `string[]`

Reply:
Got it! So the previous test should also throw if the props were declared as const right? 
```ts
const AppWithArrayProps = {
  props: ['a'] as const,
  template: ''
}
```

Reply:
That doesn't seem to happen, if you define `as const`, it will use `props: {}` overload 🤔